### PR TITLE
Fixed logs, added userError to getSecrets, added decryption failure test

### PR DESCRIPTION
**Pull request by KuphJr**

### CODE REVIEW

**Comment by KuphJr** [comment]
I added a new `decryptionRequestTimeoutSec` configuration parameter to the jobspec.  This fixes the problem where the threshold decryption hangs, but the `listenerEventHandlerTimeoutSec` is exceeded and the entire request fails before returning the threshold decryption error back to the user.

**Review by pinebit — APPROVED** [review]

File: core/services/ocr2/plugins/threshold/decryption_queue.go [hunk 1]
```diff
@@ -105,7 +105,7 @@ func (dq *decryptionQueue) getResult(ciphertextId decryptionPlugin.CiphertextId,
 
 	req, ok := dq.completedRequests[string(ciphertextId)]
 	if ok {
-		dq.lggr.Debugf("ciphertextId %s was already decrypted by the DON", ciphertextId)
+		dq.lggr.Debugf("ciphertextId %s was already decrypted by the DON", string(ciphertextId))
```
Comment (bolekk) [inline core/services/ocr2/plugins/threshold/decryption_queue.go:108]: Is ciphertext printable? Shouldn't we convert to hex before logging?
Also Debugw() is preferable.

**Review by bolekk — CHANGES_REQUESTED** [review]

File: core/services/functions/listener.go [hunk 2]
```diff
@@ -115,23 +115,24 @@ const (
...
+	serviceCancel         context.CancelFunc
+	chStop                chan struct{}
+	pluginORM             ORM
+	pluginConfig          config.PluginConfig
+	s4Storage             s4.Storage
+	logger                logger.Logger
+	mailMon               *utils.MailboxMonitor
+	urlsMonEndpoint       commontypes.MonitoringEndpoint
+	decryptor             threshold.Decryptor
+	decryptRequestTimeout time.Duration
```
[the remark this answers is not in the document]
Comment (KuphJr) [reply core/services/functions/listener.go:135]: This is a timeout specifically for decryption.  We cannot use the listenerEventHandlerTimeout or the requestTimeout as we need a specific timeout which pertains only to decryption attempts.

File: core/services/ocr2/plugins/threshold/decryption_queue.go (hunk 1 above)
Comment (KuphJr) [reply core/services/ocr2/plugins/threshold/decryption_queue.go:108]: I am already using debugf everywhere else.  Unless you have a strong preference, I think we should keep it.

Also, when looking at the logs, they appear as I would expect with the log messages as they appear in the PR.

File: core/services/functions/listener.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (bolekk) [reply core/services/functions/listener.go:135]: Sure but  it's available in `pluginConfig.DecryptionQueueConfig.DecryptRequestTimeoutSec`.

**Review by bolekk — CHANGES_REQUESTED** [review]

File: core/services/functions/listener.go [hunk 3]
```diff
@@ -522,52 +528,58 @@ func (l *FunctionsListener) reportSourceCodeDomains(requestId RequestID, domains
...
 		secrets = record.Payload
 	}
 
 	if len(secrets) == 0 {
-		return "", nil
+		return "", nil, nil
 	}
 
-	decryptedSecrets, err := l.decryptor.Decrypt(ctx, []byte(requestID), secrets)
+	fmt.Println("\n\n", ctx, time.Duration(l.pluginConfig.DecryptionQueueConfig.DecryptRequestTimeoutSec)*time.Second)
```
Comment (pinebit) [inline core/services/functions/listener.go:575]: I believe this is a leftover - fmt.Println

**Review by pinebit — APPROVED** [review]

**Review by bolekk — APPROVED** [review]

**Pull request merged**